todos-lambda/Sources/App/lambda.swift

```diff
@@ -19,53 +19,72 @@ import HummingbirdLambda
-/// You just need to implement the `buildResponder` function to handle requests.
-/// Alternatively, you can use the `APIGatewayV2LambdaFunction` conformance to build a Lambda function for APIGatewayV2.
+/// The main entry point for this app. In the main function, create a Hummingbird router
+/// using a context, and provide it to the `APIGatewayLambdaFunction`.
+///
+/// Note: This sample project uses the `@main` attribute and a dedicated type, `App`,
+/// to organize the code in a cleaner way. Alternatively, if you want a smaller approach, add the snippet in
+/// the [Hummingbird-Lambda package's
+/// readme](https://github.com/hummingbird-project/hummingbird-lambda) to the
+/// target's main.swift file instead.
 @main
-struct AppLambda: APIGatewayLambdaFunction {
-    /// The Context for your application. All Lambda functions needa LambdaRequestContext conforming type.
-    /// The default type for a Lambda function is `BasicLambdaRequestContext<APIGatewayRequest>`.
-    /// or `BasicLambdaRequestContext<APIGatewayV2Request>` for APIGatewayV2 services.
+struct App {
+    /// The Context for your application. All Lambda functions need a LambdaRequestContext conforming type.
     /// 
-    /// It's common to define a typealias for your preferred Context type. That way you can change the Context type
-    /// if you need to, and only have to change one line of code.
+    /// It's common to define a typealias for your preferred Context type. That way you can change the
+    /// Context type if you need to, and only have to change one line of code.
+    ///
+    /// **Important:** this sample project uses API Gateway to expose the function to the web, and
+    ///  and therefore the request context uses `APIGatewayRequest`. In case you use the
+    ///  simpler function URL instead, change the APIGatewayRequest in your context to
+    ///  a `FunctionURLRequest`, and the lambda to `FunctionURLLambdaFunction`.
+    ///  The same applies to API Gateway V2.
     typealias Context = BasicLambdaRequestContext<APIGatewayRequest>
-    
-    let awsClient: AWSClient
-    let logger: Logger
 
-    /// Initialized the Lambda function. This is called when the lambda function is first spun up.
-    /// We initialize the AWS client and logger here. You can also initialize any other services here.
-    /// For example, Logging, Metrics and Tracing should be set up here.
-    init(context: LambdaInitializationContext) {
+    private let awsClient: AWSClient
+    private let logger: Logger
+
+    /// Initializes the App struct, builds the router, and run the lambda using the router.
+    static func main() async throws {
+        let app = App()
+        let router = app.buildRouter()
+        let lambda = APIGatewayLambdaFunction(router: router)
+        try await lambda.runService()
+
+        // Shut down the AWS client and other services after the lambda
+        // is done
+        try await app.shutdown()
```

Comment:
This won't get called if `lambda.runService()` throws an error.
In other places where I've used Soto, I've added an AWSClientService. eg
```swift
struct AWSClientService: Service {
    let client: AWSClient

    func run() async throws {
        // Ignore cancellation error
        try? await gracefulShutdown()
        try await self.client.shutdown()
    }
}
```

Reply:
Sounds good! Let me know when it's available then.

Reply:
Its in 7.9.0